databases/business-sync/fragment-shoebox-snippet.js

```diff
@@ -0,0 +1,40 @@
+{
+  channels: getDocSyncChannels(doc, oldDoc),
+  typeFilter: function(doc, oldDoc) {
+    return createBusinessEntityRegex('shoeboxSnippet\\.[A-Za-z0-9_-]+\\.[A-Za-z0-9_-]+').test(doc._id);
+  },
+  allowUnknownProperties: true,
+  immutable: true,
+  propertyValidators: {
+    type: {
+      // The type of shoebox snippet (ie, bank transaction, email, photo, digital receipt)
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    source: {
+      // The source of the snippet
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    kashooEntityType: {
+      // Type of corresponding Books entity
+      type: 'string',
+      required: false,
+      mustNotBeEmpty: true
+    },
+    kashooId: {
+      // The ID of the corresponding Books entity
+      type: 'integer',
+      required: false,
+      minimumValue: 1
+    },
+    data: {
+      // Raw data of the snippet
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    }
```

Comment:
I'm flip-flopping on a `processed` boolean. Without it, it wouldnt be trivial to tell whether an item was already read by the system or not, it would take a query (`select item where id not in processed-items...`). This may also be inefficient (for example, for bank rec, `select all items where item.id not in processed).

On the other hand, it may be inefficient to update each shoebox item whenever it's actually processed (`update item set processed = true`...). This could be done as a bulk call to couchbase but it would still touch a bunch of random bytes throughout the table...

Any more thoughts in reply to mine?